### PR TITLE
libjulia_jll: turn LibOSXUnwind_jll into a build dep for Julia 1.6

### PR DESCRIPTION
**Pull request by fingolfin**
This is experimental. I still need to verify that the result can
be used to build other JLLs using `libjulia_jll`. But I thought
it'd be good to already collect feedback here.

[skip ci] [skip build]﻿

### CODE REVIEW

File: L/libjulia/common.jl [hunk 1]
```diff
@@ -314,7 +322,7 @@ function build_julia(ARGS, version::VersionNumber; jllversion=version)
     if version < v"1.5.1"
         push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.5"))
     elseif version < v"1.7"
-        push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.6"))
+        push!(dependencies, BuildDependency("LibOSXUnwind_jll", compat="0.0.6"))
```
Comment (giordano) [inline L/libjulia/common.jl:325]:
```suggestion
        push!(dependencies, BuildDependency(PackageSpec(; name="LibOSXUnwind_jll", version=v"0.0.6")))
```

File: L/libjulia/common.jl [hunk 2]
```diff
@@ -314,7 +322,7 @@ function build_julia(ARGS, version::VersionNumber; jllversion=version)
     if version < v"1.5.1"
         push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.5"))
     elseif version < v"1.7"
-        push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.6"))
+        push!(dependencies, BuildDependency(PackageSpec(; name="LibOSXUnwind_jll", version=v"0.0.6")))
```
Comment (giordano) [inline L/libjulia/common.jl:325]:
```suggestion
        push!(dependencies, BuildDependency(PackageSpec(; name="LibOSXUnwind_jll", uuid="a83860b7-747b-57cf-bf1f-3e79990d037f", version=v"0.0.6")))
```

**Comment by fingolfin** [comment]
Thanks for working on this @giordano. I've renamed my patch to hopefully fix the `cd: too many arguments` error. That leaves `KeyError: key UUID("a83860b7-747b-57cf-bf1f-3e79990d037f") not found` for Julia 1.6.

**Comment by giordano** [comment]
Yeah, this is becoming a circular issue...

**Comment by fingolfin** [comment]
:-(.

next idealy then: manually download and extract the libunwindOSX artifact, remove it as a builddep. But need sleep first

**Comment by fingolfin** [comment]
@giordano how can I fix this again?
```
ERROR: LoadError: Unsatisfiable requirements detected for package OpenBLAS_jll [4536629a]:
 OpenBLAS_jll [4536629a] log:
 ├─possible versions are: 0.3.13 or uninstalled
 └─restricted to versions 0.3.10 by an explicit requirement — no versions left
```

**Comment by fingolfin** [comment]
*sigh*
```
ERROR: LoadError: Unsatisfiable requirements detected for package libLLVM_jll [8f36deef]:
 libLLVM_jll [8f36deef] log:
 ├─possible versions are: 12.0.1 or uninstalled
 └─restricted to versions 11.0.1 by an explicit requirement — no versions left
```

**Comment by giordano** [comment]
Yeah, they all need to become `get_addable_spec`.  Every time we move forward with the Julia used for the build, old dependences may become uninstallable otherwise.

**Comment by fingolfin** [comment]
This should be ready. But perhaps should be merged with skip ci, skip build, and then we focus on PR #4461.

**Comment by giordano** [comment]
Thanks a lot Max!

**Pull request merged**